server/controller/todoController.ts

```diff
@@ -0,0 +1,202 @@
+import { Request, Response } from 'express';
+import TodoService from '../service/todoService';
+import UserModel from '../model/userModel';
+import { createHash } from 'crypto';
+import crypto from 'crypto';
+import jwt from 'jsonwebtoken';
+import axios from 'axios';
+import todoService from '../service/todoService';
+
```

Comment:
Can you add some validations to these functions for example, where you need to find a todo by id (or delete/update/..) and you are looking it up by id you should check if the id is provided and is not empty. That way you can throw an error before even calling the service.